prime-router/src/main/kotlin/azure/Event.kt

```diff
@@ -297,4 +273,17 @@ class BatchEvent(
     companion object {
         const val eventType = "receiver"
     }
-}
\ No newline at end of file
+}
+
+@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
+data class QueueMessage(
```

Comment:
Can we just extend the existing `Message` class?  I'm not sure why we need more than one QueueMessage class

Reply:
There is no overlap between the two (the reportId needs to be nullable), but yes, I can do this.

Reply:
Because it requires different fields and having to override the fields in the message leads to odd cases like double fields for places that need to be nullable, fields that are required that we don't have values for in that part of the code, etc.

Reply:
Right, but we could move those fields around if they're not actually generic across all our messages?  There is usefulness in having a consistent abstract class, as an example the `Message` class provides some checking if the message is too large for the queue

Reply:
I could be wrong, but this new class has all these properties because it handles multiple kinds of events.  An alternative being a data class per event class that extends from a common ancestor

Reply:
I am not talking about the fields that I created for the QueueMessage. I would need to change the base class so that each of those fields was nullable since almost all of those would be null/empty for these messages. Is that acceptable?

Reply:
If the existing fields on `Message` are not generic across all messages, they can just be moved to the subclasses that need them?    Is that the question here?